express-notes-api/index.js

```diff
@@ -0,0 +1,103 @@
+var express = require('express');
+var app = express();
+
+const data = require('./data.json');
+
+app.get('/api/notes', function (req, res) {
+  var notesArray = [];
+  for (const property in data.notes) {
+    notesArray.push(data.notes[property]);
+  };
+  res.status(200);
+  res.json(notesArray);
+});
+
+app.use(express.json());
+
+app.get('/api/notes/:id', function (req, res) {
+  var notesId = req.params.id;
+  if (Math.sign(notesId) === 1) {
+    if (data.notes[notesId] !== undefined) {
+      res.status(200);
+      res.json(data.notes[notesId]);
+    } else {
+      res.status(404);
+      res.json({ "error": `cannot find note with id ${notesId}` })
+    };
+  } else {
+    res.status(400);
+    res.json({ "error": "id must be a positive integer"})
+  };
+});
+
+app.post('/api/notes', function (req,res) {
+  var noteContent = req.body;
+  if (noteContent.content !== undefined) {
+    if (data.nextId !== undefined) {
+      data.notes[data.nextId] = noteContent;
+      data.notes[data.nextId].id = data.nextId;
+      res.status(201);
+      res.json(data.notes[data.nextId]);
+      data.nextId++;
```

Comment:
The new note data needs to be written back to the `data.json` file, otherwise the change is lost when the Express server restarts:

---

![image](https://user-images.githubusercontent.com/16143771/117694860-7c05a700-b174-11eb-9d51-b582fe4c0143.png)